include/h2o.h

```diff
@@ -467,6 +467,16 @@ enum {
 };
 
 /**
+ * holds various data related to the context
+ */
+typedef struct st_h2o_context_storage_item_t {
+    void (*dispose)(struct st_h2o_context_storage_item_t *self, h2o_context_t *ctx);
+    void *data;
+} h2o_context_storage_item_t;
+
+typedef H2O_VECTOR(h2o_context_storage_item_t *) h2o_context_storage_t;
```

Comment:
How about defining the type as `H2O_VECTOR(h2o_context_storage_item_t)`?

I believe there is no reason to separately allocate each item. Using a flat struct (opposed to list of pointers) would reduce number of allocation and CPU cycles spent for accessing the values.